rclpy/src/rclpy/serialization.cpp

```diff
@@ -0,0 +1,149 @@
+// Copyright 2021 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Include pybind11 before rclpy_common/handle.h includes Python.h
+#include <pybind11/pybind11.h>
+
+#include <rcl/error_handling.h>
+#include <rcl/rcl.h>
+#include <rcl/types.h>
+#include <rcutils/allocator.h>
+#include <rcutils/error_handling.h>
+
+#include <memory>
+#include <stdexcept>
+
+extern "C"
+{
+#include "rclpy_common/common.h"
+}
+#include "rclpy_common/handle.h"
+
+#include "rclpy_common/exceptions.hpp"
+
+#include "serialization.hpp"
+
+namespace rclpy
+{
+class SerializedMessage
+{
+public:
+  SerializedMessage()
+  {
+    rcl_msg_ = rmw_get_zero_initialized_serialized_message();
+  }
+
+  void
+  init(rcutils_allocator_t allocator)
+  {
+    rcutils_ret_t rcutils_ret = rmw_serialized_message_init(&rcl_msg_, 0u, &allocator);
+    if (RCUTILS_RET_OK != rcutils_ret) {
+      throw RCLError("failed to initialize serialized message");
+    }
+  }
+
+  ~SerializedMessage()
+  {
+    rcutils_ret_t ret = rmw_serialized_message_fini(&rcl_msg_);
+    if (RCUTILS_RET_OK != ret) {
+      fprintf(
+        stderr,
+        "[rclpy|" RCUTILS_STRINGIFY(__FILE__) ":" RCUTILS_STRINGIFY(__LINE__) "]: "
+        "failed to fini rcl_serialized_msg_t (%d) in destructor: %s\n",
+        ret,
+        rcutils_get_error_string().str);
+      rcutils_reset_error();
+    }
+  }
+
+  rcl_serialized_message_t rcl_msg_;
```

Comment:
@sloretz meta: given it's a public member, shall we drop the trailing underscore?

Reply:
I [didn't find anything in the style guide](https://google.github.io/styleguide/cppguide.html#Variable_Names) about trailing underscore's only being for private member variables, but it says structs don't have them. I converted it to a struct and removed the trailing underscore in 5fc0e07